modules/tinymce/src/core/main/ts/html/ParserFilters.ts

```diff
@@ -172,6 +173,24 @@ const register = (parser, settings: any): void => {
     });
   }
 
+  // Remove style attribute from styling elements and apply it to an inner span instead
+  parser.addNodeFilter('strong, b, em, i, u, strike', (nodes) => {
+    Arr.each(nodes, (node) => {
+      if (!node.attr('style')) {
+        return;
+      }
+      const span = Node.create('span', { style: node.attr('data-mce-style') || node.attr('style') });
+      // Insert span inside node
+      while (node.firstChild) {
```

Comment:
How is this loop ever completing?

Reply:
`firstChild` is null when there are no children. We probably shouldn't encourage this code style 🤔

Reply:
I agree that it is difficult to deduce what is happening from inspection. I'll pull it out into a function and add some more commenting to make it more clear.

Reply:
I didn't mean switching to options; even just `while (node.firstChild !== null) {` would be clearer.